readthedocs/projects/models.py

```diff
@@ -225,6 +230,35 @@ class Project(models.Model):
         ),
     )
 
+    urlpattern = models.CharField(
+        _("Custom URL pattern for multi version and single version projects"),
+        max_length=255,
+        default=None,
+        blank=True,
+        null=True,
+        help_text=_(
+            "A Python regex pattern used when evaluating a multi version or single version project. "  # noqa
```

Comment:
We were hoping to make this a bit less complex by not exposing regex here. If we need the complexity for the implementation, I'm OK with it, but don't fully understand why we need a full regex here instead of something simpler.